pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -158,9 +162,62 @@ func (c *Controller) Run(numWorkers int, stopCh <-chan struct{}, controllerManag
 		},
 	})
 
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			node := obj.(*v1.Node)
+			if c.nonDefaultParamsPodRanges(node) {
+				c.queue.Add(networkv1.DefaultPodNetworkName)
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			oldNode := old.(*v1.Node)
+			newNode := new.(*v1.Node)
+			// the node Pod CIDR could change if the node restarts
+			// controller only reacts after the node Pod CIDR is populated and the new CIDR is different from the old one
+			if newNode.Spec.PodCIDR != "" && newNode.Spec.PodCIDR != oldNode.Spec.PodCIDR {
+				if c.nonDefaultParamsPodRanges(newNode) {
+					c.queue.Add(networkv1.DefaultPodNetworkName)
+				}
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			node := obj.(*v1.Node)
+			if v, ok := node.Labels[utilnode.NodePoolPodRangeLabelPrefix]; ok {
+				if v != "" && node.Spec.PodCIDR != "" {
+					c.queue.Add(networkv1.DefaultPodNetworkName)
+				}
+			}
+		},
+	})
+
+	for _, clusterCIDR := range clusterCIDRs {
+		if netutils.IsIPv4CIDR(clusterCIDR) {
+			c.clusterDefaultIPv4PodCIDR = clusterCIDR.String()
```

Comment:
why is this convention of IPv4 only as default?

Reply:
I keep the clusterCIDR here as a list of `[]*net.IPNet` for future dual stack or single v6 use cases. 

The IPv4 CIDR is going to be used to get the cluster default Pod range name from GCE compute subnet. This Pod range name will always remain in the default GNP for the above reason: https://github.com/kubernetes/cloud-provider-gcp/pull/638#discussion_r1421123344 to prevent `params.Spec.PodIPv4Ranges = nil`.